src/index.ml

```diff
@@ -654,11 +653,10 @@ struct
               let log_key = log.(log_i).key in
               not K.(equal log_key index_key))
               && filter index_key buf_offset
-            then (
-              Bytes.blit buf buf_offset buf_str 0 Entry.encoded_size;
-              append_buf_fanout fan_out index_key_hash
-                (Bytes.unsafe_to_string buf_str)
-                dst_io);
+            then
+              append_substring_fanout fan_out index_key_hash dst_io
+                (Bytes.unsafe_to_string buf)
```

Comment:
I wanted to be sure that using `Bytes.unsafe_to_string` didn't create extra allocations and it doesn't so everything is ok for me! 

Great PR, let's merge it :-)